core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/JWTSupport.java

```diff
@@ -0,0 +1,281 @@
+				return (RSAPrivateKey) KeyFactory.getInstance("RSA").generatePrivate(keySpec);
+			} catch (InvalidKeySpecException | NoSuchAlgorithmException e) {
+				throw new SnowOwl.InitializationException("Invalid signingKey. Only Base64 encoded PKCS#1, PKCS#8 or JWK keys are supported. Error: " + e.getMessage());
+			}			
+		} else {
+			// TODO support PKCS#1
+			// TODO support JWK strings
+			throw new SnowOwl.InitializationException(String.format(""));
+		}
+	}
+
+	private RSAPublicKey readPublicKey(String value) {
+	    try {
+	    	String extractedKey = value
+					// replace header
+					.replace(PUBLIC_HEADER, "")
+					// replace any line endings if present
+					.replaceAll("\\r", "")
+					.replaceAll("\\n", "")
+					// replace footer
+					.replace(PUBLIC_FOOTER, "");
+	    	X509EncodedKeySpec keySpec = new X509EncodedKeySpec(Base64.getDecoder().decode(extractedKey));
+			return (RSAPublicKey) KeyFactory.getInstance("RSA").generatePublic(keySpec);
+		} catch (InvalidKeySpecException | NoSuchAlgorithmException e) {
+			throw new SnowOwl.InitializationException("Invalid verificationKey. Only Base64 encoded X509 Certificate keys are supported. Error: " + e.getMessage());
+		}
+	}
+	
+	public static final com.auth0.jwt.JWTVerifier createJWTVerifier(Algorithm algorithm, final JWTConfiguration conf) {
+		return JWT.require(algorithm)
+				.withIssuer(conf.getIssuer())
+				.acceptLeeway(3L) // 3 seconds
+				.build();
+	}
+	
+	public static final Algorithm throwUnsupportedJws(JWTConfiguration config, RSAKeyProvider keyProvider) {
+		throw new SnowOwl.InitializationException(String.format("Unsupported JWT token signing algorithm: %s", config.getJws()));
+	}
+	
+	/**
+	 * Converts the given JWT access token to a {@link User} representation using the configured email and permission claims.
+	 * 
+	 * @param jwt
+	 *            - the JWT to convert to a {@link User} object
+	 * @return
+	 * @throws BadRequestException
+	 *             - if either the configured email or permissions property is missing from the given JWT
+	 */
+	public static User toUser(DecodedJWT jwt, JWTConfiguration config) {
+		final String emailClaimProperty = config.getEmailClaimProperty();
+		final Claim emailClaim = jwt.getClaim(emailClaimProperty);
+		if (emailClaim == null || emailClaim.isNull()) {
+			throw new BadRequestException("'%s' JWT access token field is required for email access, but it was missing.", emailClaimProperty);
+		}
+
+		final String permissionsClaimProperty = config.getPermissionsClaimProperty();
+		Claim permissionsClaim = jwt.getClaim(permissionsClaimProperty);
+		if (permissionsClaim == null || permissionsClaim.isNull()) {
+			throw new BadRequestException("'%s' JWT access token field is required for permissions access, but it was missing.",
+					permissionsClaimProperty);
+		}
+
+		final List<Permission> permissions = jwt.getClaim(permissionsClaimProperty).asList(String.class).stream().map(Permission::valueOf)
```

Comment:
The claim already seems to be extracted to a local variable `permissionsClaim`. I'd recommend using it again in this expression like `emailClaim` is re-used below.